backend/Domain/IRepository.cs

```diff
@@ -5,6 +5,9 @@ public interface IRepository
     {
         List<User> Users { get; }
         void Add<T>(T entity) where T : class;
+        void Delete<T>(T entity) where T : class;
+        T GetById<T>(int id) where T : class;
+        bool isUsernameAvailable(string username);
```

Comment:
this is not in repository because this is domain specific. Repo is more generic.
in order to get info if username is available you need to use User list and search using Linq(Users.Where(...))